src/function/maybe_changed_after.rs

```diff
@@ -260,6 +263,33 @@ where
         true
     }
 
+    /// If this is a provisional memo, validate that it was cached in the same iteration of the
+    /// same cycle(s) that we are still executing. If so, it is valid for reuse. This avoids
+    /// runaway re-execution of the same queries within a fixpoint iteration.
+    pub(super) fn validate_same_iteration(
+        &self,
+        db: &C::DbView,
+        database_key_index: DatabaseKeyIndex,
+        memo: &Memo<C::Output<'_>>,
+    ) -> bool {
+        tracing::debug!(
+            "{database_key_index:?}: validate_same_iteration(memo = {memo:#?})",
+            memo = memo.tracing_debug()
+        );
```

Comment:
I wonder how expensive the `tracing` calls are in a function as hot as `fetch_hot`. It might be worth reducing them to `trace` (They would be removed from the release bundle in red knot)

Reply:
I did this anyway, "compiled out" seems best, and these are very in-the-weeds tracing outputs, I think having them at "trace" level is reasonable.